src/main/java/org/rxbooter/injector/core/supplier/SupplierFactory.java

```diff
@@ -140,6 +140,8 @@ private SupplierFactory() {}
 
     @SuppressWarnings("unchecked")
     private static <T> ThrowingSupplier<T> mapMethodParameters(Method method, Supplier<?>[] suppliers) {
+
+       //that is way to fugly but probably no way to not have it
         switch (method.getParameterCount()) {
```

Comment:
Definitely. This approach is result of a number of experiments and it seems simplest (and fastest) one.